simple_network_sim/sampleUseOfModel.py

```diff
@@ -58,49 +52,20 @@ def main(argv):
 def runSimulation(
     network: ss.NetworkOfPopulation,
     max_time: int,
-    trials: int,
-    initialInfections: List,
-) -> pd.DataFrame:
+) -> List[pd.DataFrame]:
     """Run pre-created network
 
     :param network: object representing the network of populations
-    :type network: A NetworkOfPopulation object
     :param max_time: Maximum time for simulation
-    :type max_time: int
-    :param trials: Number of simulation trials
-    :type trials: int
-    :param initialInfections: List of initial infection. If seeded, only one
-    :type initialInfections: list
-    :return: Averaged number of infection through time, through trials
-    :rtype: list
+    :return: Averaged number of infection through time, for all trial
     """
-    if trials <= 1:
-        # The averaging logic is slow and wastes a bunch of memory, skip it if we don't need it
-        logger.info("Running simulation (1/1)")
-        disposableNetwork = copy.deepcopy(network)
-        ss.exposeRegions(initialInfections[0], disposableNetwork.initialState)
-        return ss.basicSimulationInternalAgeStructure(disposableNetwork, max_time)
-    else:
-        aggregated = None
-
-        for i in range(trials):
-            logger.info("Running simulation (%s/%s)", i + 1, trials)
-            disposableNetwork = copy.deepcopy(network)
-
-            ss.exposeRegions(initialInfections[i], disposableNetwork.initialState)
-            indexed = ss.basicSimulationInternalAgeStructure(disposableNetwork, max_time).set_index(
-                ["time", "node", "age", "state"]
-            )
-
-            if aggregated is None:
-                aggregated = indexed
-            else:
-                aggregated.total += indexed.total
-
-        averaged = aggregated.reset_index()
-        averaged.total /= trials
+    results = []
+    for i in range(network.trials):
+        logger.info("Running simulation (%s/%s)", i + 1, network.trials)
+        result = ss.basicSimulationInternalAgeStructure(network, max_time, network.initialInfections)
+        results.append(result)
```

Comment:
Why did you stop averaging the results and saving just the first one? I think it makes sense for the parameter sensitivity analysis to see all the results, and I suppose that's why you changed the runSimulation function. But I think for the "normal" run, it makes sense to consolidate the results.

Reply:
I think for the stochastic run we want to keep only an array of the raw results in this function, and any aggregation be done in other functions (as it's not a part of running the simulation anymore). At minimum we'd need an average with confidence bands (per region, then aggregate, etc). Given the nature of the problem we might also want to look at best/worst cases etc. I think it will be enough calculations that we want to do it elsewhere.

For a deterministic run I agree it's a bit silly to return an array of size 1. I started off with:
```
if trial == 1:
    # compute it once
    return result
else:
    # compute many times
    return results
```

I can change it back to this if you prefer?

Reply:
I don't see a reason to delete the old code while we don't have a better way to aggregate the results. I feel like averaging them was a better behaviour than just saving the first result. At least that's my understanding, I may be missing something.

Reply:
agree with @aflag - suggest keeping the mean for now, adding a better aggregator later?  Reluctant to have a stochastic version without some sort of vaguely sensible reporting, even if we're going to improve it later.

Reply:
Thanks both, I've added a separate aggregation function which returns the mean of all the results. Let me know if it looks good. 
For what it's worth, from limited testing it seems the average of a few stochastic runs is very close to the deterministic run:
![image](https://user-images.githubusercontent.com/24376790/86483683-75382780-bd4c-11ea-97b2-80f396c270f3.png)